src/sif/util.cc

```diff
@@ -0,0 +1,132 @@
+#include "sif/util.h"
+#include "baldr/time_info.h"
+
+namespace valhalla {
+namespace sif {
+
+// what this function calls to get the next edge
+using EdgeCallback = std::function<baldr::GraphId(void)>;
+// what this function calls to emit the next label
+using LabelCallback = std::function<void(const EdgeLabel& label)>;
+
+/**
+ * Will take a sequence of edges and create the set of edge labels that would represent it
+ * Allows for the caller to essentially re-compute the costing of a given path
+ *
+ * @param reader            used to get access to graph data. modifyable because its got a cache
+ * @param costing           single costing object to be used for costing/access computations
+ * @param edge_cb           the callback used to get each edge in the path
+ * @param label_cb          the callback used to emit each label in the path
+ * @param source_pct        the percent along the initial edge the source location is
+ * @param target_pct        the percent along the final edge the target location is
+ * @param date_time         the time string representing the local time before traversing the first
+ *                          edge. of the format YYYY-MM-DDTHH:mm
+ */
+void recost_forward(baldr::GraphReader& reader,
```

Comment:
this is the main function we use to walk a path with a different costing to get its different cost info